pkg/agent/route/route_linux.go

```diff
@@ -772,3 +775,124 @@ func (c *Client) DeleteSNATRule(mark uint32) error {
 	snatIP := value.(net.IP)
 	return c.ipt.DeleteRule(iptables.NATTable, antreaPostRoutingChain, c.snatRuleSpec(snatIP, mark))
 }
+
+func (c *Client) AddNodePortConfig(nodePortIPMap map[int][]net.IP) error {
+	gatewayIfIndex := getIndexByName(c.nodeConfig.GatewayConfig.Name)
+	// Clean left TC config.
+	qdiscsClean(nodePortIPMap, gatewayIfIndex)
```

Comment:
Will this cause any traffic disruption? Will it impact existing connections?

Reply:
TC config generated by AntreaProxy will not be removed when uninstalling Antrea. When installing Antrea once again, this left TC config may impact the installation. `qdiscsClean` can clean left TC config. 

Another situation is that before installing Antrea, there is some TC config being used by existing connections. In this case, `qdiscsClean` will impact existing connections.

Reply:
I thought the func will be called every time Agent restarts? If so will Agent restart cause disruption? Will existing connections be broken?

Reply:
This func will be called every time when Agent restarts. When restarting Agent, existing connections (e.g NodePort Service connections) matched by TC filters will be broken.

Reply:
This sounds pretty bad, and so far Agent restart will not break any existing connection. Any way to keep the TC state at Agent restarts?

Reply:
It seems that I misunderstood the meaning of `break existing connection`. When restarting Agent, TC config will be cleaned and rebuilt. At the moment of restarting, there should be packet loss for existing connection.

Reply:
That sounds much better! But do you know how long the packet loss can be?

Reply:
We should minimize packet loss.

Reply:
> We should minimize packet loss.
Agreed.

> That sounds much better! But do you know how long the packet loss can be?
I think I can make a test to verify the packet loss.

Reply:
That will be great. Please add some comments to the code about the packet loss and the estimated time.